kmk/modules/pimoroni_trackball.py

```diff
@@ -48,76 +48,158 @@
 ANGLE_OFFSET = 0
 
 
+class TrackballHandlerKeyMeta:
+    def __init__(self, handler=0):
+        self.handler = handler
+
+
+def layer_key_validator(handler):
+    return TrackballHandlerKeyMeta(handler=handler)
+
+
 class TrackballMode:
-    '''Behaviour mode of trackball: mouse movement or vertical scroll'''
+    """Behaviour mode of trackball: mouse movement or vertical scroll"""
 
     MOUSE_MODE = const(0)
     SCROLL_MODE = const(1)
 
 
-class Trackball(Module):
-    '''Module handles usage of Trackball Breakout by Pimoroni'''
+class TrackballHandler:
+    def handle(self, keyboard, trackball, x, y, switch, state):
+        raise NotImplementedError
+
+
+class PointingHandler(TrackballHandler):
+    def handle(self, keyboard, trackball, x, y, switch, state):
+        if trackball.mode == TrackballMode.MOUSE_MODE:
+            if x >= 0:
+                trackball.pointing_device.report_x[0] = x
+            else:
+                trackball.pointing_device.report_x[0] = 0xFF & x
+            if y >= 0:
+                trackball.pointing_device.report_y[0] = y
+            else:
+                trackball.pointing_device.report_y[0] = 0xFF & y
+            trackball.pointing_device.hid_pending = x != 0 or y != 0
+        else:  # SCROLL_MODE
```

Comment:
`SCROLL_MODE` is essentially just another `handler` and it would be nice to have a single mechanism to set modes / handlers. We can either implement this in this PR, or open an issue for a future improvement, whichever you favour.

Reply:
the latest commit does that: adding a seperate handler for scroll, and deprecate the old parameter while providing a sane fallback